### PR TITLE
doc and error for existing non-appendable target

### LINKED ISSUE #283: can not use an existing DataFrame as target in odo(source, target)
It seems an existing DataFrame can't be used as target, even when its datashape matches source, not sure if this is intentional, or a bug?

for example:
```
csv = resource('test.csv')
df = odo(csv, pd.DataFrame) #initialize same as csv
odo(df, csv) #fine, simply append df content to csv
odo(csv, df) #throws NotImplementedError 
```
among csv, list, db (sqlite), they all can convert (append) between each other without issue, but none of them can be converted (append) to existing DataFrame. 
Specifically:
self appending -- odo(x,x) works fine for csv, list, sqlite etc, but
odo(df, df) does not work.

### PR DESCRIPTION
**Pull request by ywang007**
close #283 

I added the ValueError and doc for existing target that does not support in-place append. Please check.

Is there a better way than enumerate DataFrame, ndarray etc?

### CODE REVIEW

File: odo/into.py [hunk 1]
```diff
@@ -94,6 +96,8 @@ def into_object(target, source, dshape=None, **kwargs):
     """
     if isinstance(source, (str, unicode)):
         source = resource(source, dshape=dshape, **kwargs)
+    if isinstance(target, (DataFrame, Series, ndarray)):
```
Comment (cpcloud) [inline odo/into.py]: suggest a global variable like this:

```python
not_appendable = DataFrame, Series, ndarray, tuple

...
    if isinstance(target, not_appendable):
        raise TypeError('target of type %r does not support in-place append' % type(target).__name__)
```

**Comment by llllllllll** [comment]
closing and reopening to see the test output. looks like it is pointing to the old name,

**Comment by ywang007** [comment]
@llllllllll which "old name"?   Some tests which ran OK before suddenly fail now, was it some environment/software version change? Do you have a solution for that? Thanks.

**Comment by llllllllll** [comment]
sorry, the link for the test was incorrect based on the organisation change from continuumio to blaze.

The tests are failing because `EmptyHDFStoreDataset` is a subclass of tuple.

```python
EmptyHDFStoreDataset = namedtuple('EmptyHDFStoreDataset', 'parent,pathname,dshape')
```
I am wondering if the cost of maintaining this set of unappendables is worth the slightly better error message. Subtle things like this could crop up in user defined types. What if a user is dispatching on their own subclasses of these types?

**Comment by ywang007** [comment]
Yes, checking "type(target) in not_appendable_types” seems better. Even though it let pass those subclass which is indeed still not-appendable, at least it won’t raise false alarm and fail when the target is good.

File: odo/tests/test_into.py [hunk 2]
```diff
@@ -20,6 +20,11 @@ def test_into_append():
     assert result is lst
 
 
+def test_into_append_failure():
+    nd = np.array([1, 2, 3])
+    assert raises(ValueError, lambda: odo((4, 5), nd))
```
Comment (cpcloud) [inline odo/tests/test_into.py]: can you write this as

```python
```

File: odo/into.py [hunk 3]
```diff
@@ -94,6 +97,8 @@ def into_object(target, source, dshape=None, **kwargs):
     """
     if isinstance(source, (str, unicode)):
         source = resource(source, dshape=dshape, **kwargs)
+    if type(target) in not_appendable_types:
+        raise ValueError('target of %s type does not support in-place append' % type(target))
```
Comment (cpcloud) [inline odo/into.py]: I think this should be a `TypeError`

File: docs/source/overview.rst [hunk 4]
```diff
@@ -43,6 +43,12 @@ So the following lines would be valid inputs to ``odo``
    >>> odo('postgresql://hostname::tablename', 'myfile.json') # Dump Postgres to JSON
    >>> odo('mongodb://hostname/db::collection', pd.DataFrame) # Dump Mongo to DataFrame
 
+Please note that, if target already exists, it must be of a type that 
+supports in-place append.
+
+.. code-block:: python
+
+   >>> odo('myfile.csv', df) # this will raise ValueError because DataFrame is not appendable
```
Comment (cpcloud) [inline docs/source/overview.rst]: `TypeError`

**Comment by cpcloud** [comment]
@ywang007 couple of small comments, other than those good 2 go!

**Comment by ywang007** [comment]
@cpcloud changed per your comments. Thanks.

File: odo/tests/test_into.py [hunk 5]
```diff
@@ -20,6 +20,12 @@ def test_into_append():
     assert result is lst
 
 
+def test_into_append_failure():
+    with pytest.raises(TypeError):
+        nd = np.array([1, 2, 3])
```
Comment (cpcloud) [inline odo/tests/test_into.py]: keep this outside of the `with` statement, as we only want to catch errors triggered by the `odo` call

File: odo/tests/test_into.py [hunk 6]
```diff
@@ -48,3 +54,5 @@ def test_into_string_dshape():
     np.testing.assert_array_equal(odo([1, 2, 3], np.ndarray,
                                       dshape='var * float64'),
                                   np.array([1, 2, 3], dtype='float64'))
+
+test_into_append_failure()
```
Comment (cpcloud) [inline odo/tests/test_into.py]: don't call this explicitly, pytest does that for you

**Comment by cpcloud** [comment]
@ywang007 thanks! this is in

**Pull request merged**